src/forms/Button.jsx

```diff
@@ -13,29 +24,69 @@ class Button extends React.PureComponent {
 
 	render() {
 		const {
-			// removing reset & fullWidth props from other to prevent an invalid boolean being passed to <button>
-			fullWidth, // eslint-disable-line no-unused-vars
+			fullWidth,
 			icon,
 			hasHoverShadow, // eslint-disable-line no-unused-vars
 			component,
+			to,
 			...other
 		} = this.props;
 
 		// checking for the icon component signature if passed as icon={<Icon shape="search"/>} and grabbing the shape prop
-		// since this functionality is built into button in order to enforce strict sizes on button icons
-		const props = { icon: icon && icon.props && icon.props.shape };
+		const iconShape = icon && icon.props && icon.props.shape;
+
+		// this.props.component is either a string or a Link component with a `component.name` prop
+		const componentName = component
```

Comment:
nested ternaries are hard to reason about, consider this option to make the logic explicit

```js
const componentName = component === Link ? 'Link' : component;
```

Reply:
`component === Link` unfortunately doesn't work. It'll always return false. Leaving as if for now - this should be a temporary solution anyhow. We need to add this support into swarm-ui and remove from here in v2.